src/main/java/org/wise/portal/presentation/web/controllers/teacher/TeacherAPIController.java

```diff
@@ -127,6 +127,10 @@ ResponseEntity<Map<String, Object>> createTeacherAccount(
       @RequestBody Map<String, String> teacherFields, HttpServletRequest request)
       throws DuplicateUsernameException, InvalidNameException {
     TeacherUserDetails tud = new TeacherUserDetails();
```

Comment:
You can move `TeacherUserDetails tud = new TeacherUserDetails();` down to just before the line `tud.setFirstname(firstName);` since we don't use that variable until then.